doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,239 @@
+
+- **Job**: A Ray application that will be submitted to a Ray cluster for execution. Once a job is submitted, it runs once on the cluster to completion or failure. Retries or different runs with different parameters should be handled by the submitter. Jobs are scoped to the lifetime of a ray cluster.
+
+- **Job Manager**: An entity that manages the lifecycle of a Job and potentially multiple ray clusters, such as scheduling, killing, polling status, getting logs, and persisting inputs / outputs. Should be highly available (HA) by default. Can be any framework with these abilities, such as Airflow.
+
+Within the context of job submission, the packaging part is equivalent to :ref:`Runtime Environments<runtime-environments>`, where we can dynamically configure your desirable ray cluster, actor or task level runtime environment for your submitted job.
+
+**Key inputs for job submission**
+
+- **Entrypoint**: Shell command to execute once your code is unpackaged with runtime environment configured
+    - Typically :code:`python your_script.py`, can also be any shell script such as :code:`echo hello`.
+- **Runtime Environment**:
+    - :code:`working_dir` as local directory: It will be automatically zipped and uploaded to target ray cluster, then unpacked to where your submitted application runs.
+    - :code:`working_dir` as remote URIs, such as s3, git or others, it will be downloaded and unpacked to where your submitted application runs. For details, see :ref:`Runtime Environments<runtime-environments>` for details.
+
+.. note::
+  **The goal of Ray job submssion is to provide a lightweight mechanism for user to submit their locally developed and tested application to a running local / remote ray cluster, thus enabling the user to package, deploy, and manage their ray application as jobs. These jobs can be submitted by a job manager of their choice.**
+
+Example - Setup
+---------------
+
+Let's start with a sample ray script as example for job submission. Once executed locally, this script will use ray APIs to print counter value of a remote actor from 1 to 5, and print the version of 'requests' module it's using.
+
+We can put this file in a local directory of your choice, with filename "script.py", so your working directory will look like:
+
+.. code-block:: bash
+
+  | your_working_directory ("./")
+  | ├── script.py
+
+.. code-block:: python
+
+    import ray
+    import requests
+
+    ray.init()
+
+    @ray.remote
+    class Counter:
+        def __init__(self):
+            self.counter = 0
+
+        def inc(self):
+            self.counter += 1
+
+        def get_counter(self):
+            return self.counter
+
+    counter = Counter.remote()
+
+    for _ in range(5):
+        ray.get(counter.inc.remote())
+        print(ray.get(counter.get_counter.remote()))
+
+    print(requests.__version__)
+
+Ray Job SDK
+------------
+
+Ray job sdk is the recommended way to submit jobs programmatically.
+
+| Start a local ray cluster headnode
```

Comment:
```suggestion
| Start a local ray cluster head node
```

Reply:
I think we should move this to the "setup" section so that we don't have to repeat it 3 times for SDK, HTTP & CLI and also because it is not part of the submission (it is part of the setup actually)